docs/DeveloperGuide.md

```diff
@@ -309,11 +331,339 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+* *Guarantees:* person record(s) is/are edited on the system
+
+**MSS**
+
+1.  User requests a filtered list of persons ([UC06](#use-case-uc06---find-persons-by-criterias)).
+2.  ResearchRoster shows the filtered list of persons.
+3.  User requests to apply a change to all records in the list.
+4.  ResearchRoster applies change to all records in the list.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The change is invalid.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+
+#### **Use Case: UC05 - List all persons**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* list of person record(s) is shown
+
+1.  User adds new person(s) to system ([UC01](#use-case-uc01---add-a-new-person)).
+2.  User requests to list persons.
+3.  ResearchRoster shows a list of persons.
+
+    Use case ends.
+
+**Extensions**
+
+* 3a. No existing person record.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case ends.
+
+
+#### **Use Case: UC06 - Find persons by criteria(s)**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* list of person record(s) that match the criteria(s) is shown
+
+**MSS**
+
+1.  User adds new person(s) to system ([UC01](#use-case-uc01---add-a-new-person)).
+2.  User requests to list persons that match criteria(s).
+3.  ResearchRoster shows the list of persons.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. No criteria is provided.
+
+    * 2a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 1.
+
+* 3a. The keyword matches no person records.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case ends.
+
+
+#### **Use Case: UC07 - Add study group tag(s) to person**
+* *Preconditions:* user has added person(s) previously
+* *Guarantees:* study group tag(s) is/are added to person record
+
+**MSS**
+
+1.  User requests a list of persons ([UC05](#use-case-uc05---list-all-persons), [UC06](#use-case-uc06---find-persons-by-criterias)).
+2.  ResearchRoster shows a list of persons.
+3.  User requests to add a study group tag(s) to a person's record.
+4.  ResearchRoster adds the tag to the person's record.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. The given input is missing tag(s) to add.
+
+    * 3b1. ResearchRoster shows an error message.
+
+      Use case resumes at step 2.
+
+
+#### **Use Case: UC08 - Remove study group tag(s) from person**
+* *Preconditions:* user has added person(s) previously, user has added study group tag(s) to person record previously
+* *Guarantees:* study group tag(s) is/are removed from person record
+
+**MSS**
+
+1.  User adds study group tag(s) to person ([UC07](#use-case-uc07---add-study-group-tags-to-person)).
```

Comment:
similar question as uc05 and uc06 regarding precondition existing but step one saying the action of the of doing what u stated in that condition